props/saber_fett263_buttons.h

```diff
@@ -285,47 +440,1090 @@ EFFECT(faston); // for EFFECT_FAST_ON
+        char ig[10];
+        itoa(ignite_time_, ig, 10);
+        current_preset_.SetStyle(blade_num_,style_parser.SetArgument(current_preset_.GetStyle(blade_num_), 19, ig));
+        break;
+      default:
+        break;
+    }
+    current_preset_.Save();
+    UpdateFont(current_preset_.preset_num);
+  }
+
+  void SaveStyleSetting() {
+    char style_arg[10];
+    itoa(calc_, style_arg, 10);
+    current_preset_.SetStyle(blade_num_,style_parser.SetArgument(current_preset_.GetStyle(blade_num_), set_num_ + 2, style_arg));
+    current_preset_.Save();
+    PlayMenuSound("msave.wav");
+    switch (set_num_) {
+      case 21:
+        show_clash_location_.Stop(blade_num_);
+        break;
+      case 22:
+        show_drag_size_.Stop(blade_num_);
+        break;
+      case 23:
+        show_emitter_size_.Stop(blade_num_);
+      default:
+        break;
+    }
+    UpdateStyle(current_preset_.preset_num);
+    menu_type_ = MENU_STYLE_SETTING_SUB;
+    announce_menu_ = true;
+  }
+
+  void CancelStyleSetting() {
+    PlayMenuSound("mcancel.wav");
+    switch (set_num_) {
+      case 21:
+        show_clash_location_.Stop(blade_num_);
+        break;
+      case 22:
+        show_drag_size_.Stop(blade_num_);
+        break;
+      case 23:
+        show_emitter_size_.Stop(blade_num_);
+        break;
+      default:
+        break;
+    }
+    UpdateStyle(current_preset_.preset_num);
+    menu_type_ = MENU_STYLE_SETTING_SUB;
+    announce_menu_ = true;
+  }
+#endif
+
+  void NewColor(int blade, int effect) {
+    char new_color[32];
+    switch (color_mode_) {
+      case COLOR_LIST:
+      case CC_COLOR_LIST:
+        itoa(Color16(color_list_[dial_]).r, new_color, 10);
+        strcat(new_color, ",");
+        itoa(Color16(color_list_[dial_]).g, new_color + strlen(new_color), 10);
+        strcat(new_color, ",");
+        itoa(Color16(color_list_[dial_]).b, new_color + strlen(new_color), 10);
+        break;
+      default:
+        itoa(Color16(hsl).r, new_color, 10);
+        strcat(new_color, ",");
+        itoa(Color16(hsl).g, new_color + strlen(new_color), 10);
+        strcat(new_color, ",");
+        itoa(Color16(hsl).b, new_color + strlen(new_color), 10);
+        break;
+    }
+    current_preset_.SetStyle(blade,style_parser.SetArgument(current_preset_.GetStyle(blade), effect + 2, new_color));
+    if (color_mode_ != CC_COLOR_LIST) color_mode_ = NONE;
+  }
+
+  void Setup() override {
+    RestoreGestureState();
+  }
+
+#ifdef FETT263_SAVE_CHOREOGRAPHY
+  bool chdir(const char* dir) override {
+    bool ret = PropBase::chdir(dir);
+    RestoreChoreo();
+    clash_count_ = 0;
+    return ret;
+  }
+#endif
+
+    enum PlayNumber {
+    PLAY_NONE,
+    PLAY_FIRST,
+    PLAY_POINT,
+    PLAY_HUNDRED,
+    PLAY_THOUSAND,
+    PLAY_SECOND,
+    PLAY_THIRD,
+    PLAY_VOLTS,
+  };
+
+  enum SayType {
+    SAY_NONE,
+    SAY_BATTERY,
+    SAY_DECIMAL,
+    SAY_DIGIT,
+    SAY_HUNDRED,
+    SAY_THOUSAND,
+  };
+
+  void SayNumber (float number) {
```

Comment:
This function is confusing.
I think maybe the say type should be an argument to this function?
I assume this is the function you call to initiate speaking a number?

Reply:
SayNumber() function updated